### PR TITLE
GAT example: lower l2_reg so the network can learn

### PR DESCRIPTION
**Pull request by cgarciae**

### CODE REVIEW

**Comment by danielegrattarola** [comment]
Hi, thanks for the PR. 

Do you know by how much this change improves performance? 
Because 5e-4 is the value reported in the original GAT paper, and if you're having trouble with reproducing results it might be necessary to change the architecture.

**Comment by cgarciae** [comment]
@danielegrattarola The current L2 penalty is causes the network to prioritize lowering the value of the weights, the training accuracy initially increases but quickly drops to around 14%, I've ran the experiment many times. I doubt this has anything to do with my machine but you could also run it on your machine just to make sure.

**Comment by danielegrattarola** [comment]
I'll have to check, because this should not happen at all. It might be symptom of some problem with the architecture. 
I'll leave this PR open until I have made sure that everything works as expected, besides the L2 rate.

**Comment by danielegrattarola** [comment]
Update: I have checked the architecture three times and it's exactly as it supposed to be. 

This drop in performance is probably related to differences between TF1 and TF2, or even between Keras and tf.keras. I have seen things like this in the past, where even minor assumptions made in the backend of frameworks can impact training significantly. 

Lowering the l2 rate seems like a good solution, so I'm merging the PR as is. Thanks for noticing the issue!

Cheers, 
D
